File: apps/ios/ScribbleFit/ScribbleFit/Features/Analytics/Domain/Models/ExerciseInsight.swift
import Foundation

public struct ExerciseInsight: Codable, Sendable {
    public let exerciseId: String
    public let estimated1RM: Double
    public let prDetected: Bool
    public let trendDirection: InsightTrend
    public let breakdownText: String
    public let timestamp: Date
    
    public init(
        exerciseId: String,
        estimated1RM: Double,
        prDetected: Bool,
        trendDirection: InsightTrend,
        breakdownText: String,
        timestamp: Date
    ) {
        self.exerciseId = exerciseId
        self.estimated1RM = estimated1RM
        self.prDetected = prDetected
        self.trendDirection = trendDirection
        self.breakdownText = breakdownText
        self.timestamp = timestamp
    }
}

public enum InsightTrend: String, Codable, Sendable {
    case improving
    case stable
    case plateaued
    case declining
}

File: apps/android/core/ai/src/main/java/com/scribblefit/core/ai/model/ExerciseInsight.kt
package com.scribblefit.core.ai.model

data class ExerciseInsight(
    val exerciseId: String,
    val estimated1RM: Double,
    val prDetected: Boolean,
    val trendDirection: InsightTrend,
    val breakdownText: String,
    val timestamp: Long
)

enum class InsightTrend {
    IMPROVING, STABLE, PLATEAUED, DECLINING
}
